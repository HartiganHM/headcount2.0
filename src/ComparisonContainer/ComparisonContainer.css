@import url("https://fonts.googleapis.com/css?family=Open+Sans|Roboto+Slab:700");
.comparison-container {
  display: flex;
  justify-content: center; }
  .comparison-container .district {
    border: 1px dashed #6D6E71;
    border-radius: 1rem;
    height: 18rem;
    overflow: hidden;
    margin: 1rem;
    text-align: center;
    max-width: 12rem;
    min-width: 12rem; }
