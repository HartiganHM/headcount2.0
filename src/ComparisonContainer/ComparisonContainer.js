import React from 'react';
import Card from '../Card/Card';
import ComparisonCard from '../ComparisonCard/ComparisonCard';
import PropTypes from 'prop-types';
import './ComparisonContainer.css';

const ComparisonContainer = ({ selectedArray, comparedData, removeCard }) => {
    let type = 'card selected';
    let placeHolder1 = selectedArray[0] ? '' : 'district';
    let placeHolder2 = selectedArray[1] ? '' : 'district';

    const selectedCards = selectedArray.map( (card) => {
        return (
            <Card district={card.location} data={card.data} key={card.location} clickFunction={removeCard} type={type}/>
        )
    })

    return (
        <div className='comparison-container'>
            <span className={placeHolder1}>{selectedCards[0] || ''}</span>
            <ComparisonCard comparedData={comparedData}/>
            <span className={placeHolder2}>{selectedCards[1] || ''}</span>
        </div>
    )
};

ComparisonContainer.propTypes = {
    selectedArray: PropTypes.array,
    comparedData: PropTypes.object,
    removeCard: PropTypes.func,
};

export default ComparisonContainer;