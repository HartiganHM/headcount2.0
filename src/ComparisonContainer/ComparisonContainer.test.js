import React from 'react';
import ReactDOM from 'react-dom';
import ComparisonContainer from './ComparisonContainer';
import { shallow, mount, render } from 'enzyme';

describe('ComparisonContainer Tests', () => {
    let renderedComparisonContainer;
    let mockObject;
    let mockArray;
    let mockFunc;

    beforeEach( () => {
        mockArray = [];
        mockObject = {};
        renderedComparisonContainer = shallow(<ComparisonContainer
                                        selectedArray={mockArray}
                                        comparedData={mockObject}
                                    />)
    })

    it('Comparison Container should exist', () => {
        expect(renderedComparisonContainer).toBeDefined();
    });

    it('Comparison Container should match the snapshot', () => {
        expect(renderedComparisonContainer).toMatchSnapshot();
    });

    it('Comparison Container should have no Cards if array is empty', () => {
        expect(renderedComparisonContainer.find('Card').length).toEqual(0);
    });

    it('Comparison Container should two Cards if array is full', () => {
        mockArray = [ {}, {} ];
        renderedComparisonContainer = shallow(<ComparisonContainer selectedArray={mockArray} />)
        console.log(renderedComparisonContainer.debug())
        expect(renderedComparisonContainer.find('Card').length).toEqual(2);
    });
})