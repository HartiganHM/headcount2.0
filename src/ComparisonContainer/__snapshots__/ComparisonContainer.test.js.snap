exports[`Comparison Container Tests Comparison Container should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComparisonContainer
    comparedData={Object {}}
    selectedArray={Array []} />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="district" />,
        <ComparisonCard
          comparedData={Object {}} />,
        <span
          className="district" />,
      ],
      "className": "comparison-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "district",
        },
        "ref": null,
        "rendered": "",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "comparedData": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "district",
        },
        "ref": null,
        "rendered": "",
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="district" />,
          <ComparisonCard
            comparedData={Object {}} />,
          <span
            className="district" />,
        ],
        "className": "comparison-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "district",
          },
          "ref": null,
          "rendered": "",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "comparedData": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "district",
          },
          "ref": null,
          "rendered": "",
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
